packages/physics-physx/src/PhysXPhysicsManager.ts

```diff
@@ -0,0 +1,234 @@
+
+  private _pxScene: any;
+
+  private readonly _onContactBegin?: Function;
+  private readonly _onContactEnd?: Function;
+  private readonly _onContactPersist?: Function;
+  private readonly _onTriggerBegin?: Function;
+  private readonly _onTriggerEnd?: Function;
+  private readonly _onTriggerPersist?: Function;
+
+  private _eventMap: Map<number, [number, PhysicsState]> = new Map();
+
+  private _gravity: Vector3 = new Vector3(0, -9.81, 0);
+
+  /** Global gravity in the physical scene */
+  get gravity(): Vector3 {
+    return this._gravity;
+  }
+
+  set gravity(value: Vector3) {
+    this._gravity = value;
+    this._pxScene.setGravity({ x: value.x, y: value.y, z: value.z });
+  }
+
+  constructor(
+    onContactBegin?: Function,
+    onContactEnd?: Function,
+    onContactPersist?: Function,
+    onTriggerBegin?: Function,
+    onTriggerEnd?: Function,
+    onTriggerPersist?: Function
+  ) {
+    this._onContactBegin = onContactBegin;
+    this._onContactEnd = onContactEnd;
+    this._onContactPersist = onContactPersist;
+    this._onTriggerBegin = onTriggerBegin;
+    this._onTriggerEnd = onTriggerEnd;
+    this._onTriggerPersist = onTriggerPersist;
+
+    const triggerCallback = {
+      onContactBegin: (obj1, obj2) => {},
+      onContactEnd: (obj1, obj2) => {},
+      onContactPersist: (obj1, obj2) => {},
+      onTriggerBegin: (obj1, obj2) => {
+        this._eventMap.set(obj1.getQueryFilterData().word0, [
+          obj2.getQueryFilterData().word0,
+          PhysicsState.TOUCH_FOUND
+        ]);
+        this._eventMap.set(obj2.getQueryFilterData().word0, [
+          obj1.getQueryFilterData().word0,
+          PhysicsState.TOUCH_FOUND
+        ]);
+      },
+      onTriggerEnd: (obj1, obj2) => {
+        this._eventMap.set(obj1.getQueryFilterData().word0, [obj2.getQueryFilterData().word0, PhysicsState.TOUCH_LOST]);
+        this._eventMap.set(obj2.getQueryFilterData().word0, [obj1.getQueryFilterData().word0, PhysicsState.TOUCH_LOST]);
+      }
+    };
+
+    const PHYSXSimulationCallbackInstance = PhysXManager.PhysX.PxSimulationEventCallback.implement(triggerCallback);
+    const sceneDesc = PhysXManager.PhysX.getDefaultSceneDesc(
+      PhysXManager.physics.getTolerancesScale(),
+      0,
+      PHYSXSimulationCallbackInstance
+    );
+    this._pxScene = PhysXManager.physics.createScene(sceneDesc);
+
+    PhysXPhysicsManager._pxRaycastHit = new PhysXManager.PhysX.PxRaycastHit();
+    PhysXPhysicsManager._pxFilterData = new PhysXManager.PhysX.PxQueryFilterData();
+  }
+
+  //--------------public APIs--------------------------------------------------
+  /**
+   * add Collider into the manager
+   * @param collider StaticCollider or DynamicCollider.
+   */
+  addCollider(collider: Collider) {
+    this._pxScene.addActor(collider._pxActor, null);
+    for (let i = 0, len = collider._shapes.length; i < len; i++) {
+      const shape = collider._shapes[i];
+      this._eventMap.set(shape._id, [null, PhysicsState.TOUCH_NONE]);
```

Comment:
this._eventMap.set(collider._shapes[i]._id, [null, PhysicsState.TOUCH_NONE]);

Reply:
In most cases, is there only one shape in the collider, if not, _eventMap can be cached.

Reply:
done, there are many shapes in collider